### PR TITLE
Delete users and identities by their owner label value

### PR DESCRIPTION
**Pull request by sbryzak**
This PR modifies the UserAccountController to delete user and identity resources based on their owner label, instead of looking them up by name.

Fixes https://issues.redhat.com/browse/CRT-1446

e2e tests: https://github.com/codeready-toolchain/toolchain-e2e/pull/492

Note: additional e2e tests to come, the current e2e test PR simply fixes the UserID encoding changes.

### CODE REVIEW

File: controllers/useraccount/useraccount_controller.go [hunk 1]
```diff
@@ -506,66 +506,56 @@ func (r *Reconciler) deleteIdentityAndUser(logger logr.Logger, config membercfg.
...
 	}
 
-	logger.Info("deleting the User resource")
-	// Delete User associated with UserAccount
-	if err := r.Client.Delete(context.TODO(), user); err != nil {
-		if !errors.IsNotFound(err) {
+	logger.Info("deleting the User resources")
+	for _, user := range userList.Items {
+		// Delete User associated with UserAccount
+		if err := r.Client.Delete(context.TODO(), &user); err != nil {
```
Comment (alexeykazakov) [inline controllers/useraccount/useraccount_controller.go:527]: This controller watches users, so shouldn't we delete one user form the list and then return? The deletion of the user will trigger another reconcile.
We should not delete multiple Users (or Identities) in the same reconcile loop.

File: controllers/useraccount/useraccount_controller.go [hunk 2]
```diff
@@ -506,66 +506,56 @@ func (r *Reconciler) deleteIdentityAndUser(logger logr.Logger, config membercfg.
...
 	}
 
-	logger.Info("deleting the Identity resource")
-	// Delete Identity associated with UserAccount
-	if err := r.Client.Delete(context.TODO(), identity); err != nil {
-		if !errors.IsNotFound(err) {
+	logger.Info("deleting the Identity resources")
+	for _, identity := range identityList.Items {
+		// Delete Identity associated with UserAccount
+		if err := r.Client.Delete(context.TODO(), &identity); err != nil {
```
Comment (alexeykazakov) [inline controllers/useraccount/useraccount_controller.go:553]: Same here.

Comment (sbryzak) [reply controllers/useraccount/useraccount_controller.go]: Done

**Review by alexeykazakov — COMMENTED** [review]
Looks good! But can you please add the missing unit tests? See the codecov report.

**Review by alexeykazakov — APPROVED** [review]
I'm approving this PR. But please take a look at the missing unit tests. It's OK to add them in the next PR though.

**Comment by MatousJobanek** [comment]
@sbryzak I see that you decided to merge the PR without adding the unit tests, but please don't forget to add the missing unit tests in the next PR!

**Pull request merged**